doc/tutorials/understanding-error-messages.md

```diff
@@ -0,0 +1,161 @@
+```
+
+If you want additional information about the mountpoint or the configfile which
+were affected by the error you can add the verbose argument `-v`:
+
+```
+Mountpoint: system/tests/hosts
+Configfile: /etc/hosts.13163:1575216150.11522.tmp
+```
+
+If you need more information you can pass the debug argument `-d` to the command which also shows at which line in the source code
+the error was emitted:
+
+```
+At: <path>/src/plugins/network/network.c:184
+```
+
+Error messages from tools (see [kdb.md](/doc/help/kdb.md)) may yield different messages such as when providing false command line arguments
+or mounting without incorrect permissions.
+
+Elektra's error message also includes the information about the `module` which emits the error indicated by the first line:
+`` Sorry, module `MODULE` issued [error|warning] `NR`: ``. The module indicates either indicates the core of elektra (`kdb`) or can be
+a concrete plugin. All plugins have to have a name and this is guaranteed by the compiler. Before Elektra v9.0 we also
+had even more fields to display such as `ingroup` that indicated which group emitted the error (e.g., plugins). In the effort to keep the
+message as succinct as possible we decided to [remove this field](/doc/decisions/ingroup_removal.md).
+
+## Error handling for developers
+
+If you want to contribute to Elektra or write your own specific plugin you can use the integrated error macros which
+eases the error handling. In general the [plugins README](plugins.md) is a good way to start you journey through writing your own plugin.
+All macros can be used by including `kdberrors.h` into the source file.
+For more information about the macros please read through `elektra_set_concrete_type_error` section in the [plugins.md](plugins.md) tutorial.
+Internally when Elektra emits an error it sets a [metadata](/doc/dev/metadata.md) to a Key which then is later on parsed into the error message.
+The key name of the error key is the mountpoint of the configuration. If an error occurs, 7 metadata are set which can be seen in our
+[error-handling tutorial](../dev/error-handling.md). Later on those metadata are passed into a message template which renders the message
+as seen above. The same goes for warnings except that there can be up to 100 warnings but only a single error.
+
+One of the major painpoints we had during refactoring of all error messages came from inconsistent messages. We encourage
+developers to read through our [error-message guidelines](/doc/dev/error-message.md). For out of memory errors we even enforced
+a consistent error message by altering the macro which is used for emitting errors. It does not contain a text field anymore because
+these messages should be identical in all cases. Each error message category comes with associated information which should be present.
+Validation errors for example should require the key and value in every message but is not "templated" yet. If you cannot find good possibility
```

Comment:
I think this text does not help the reader of the tutorial. Ideally you would finish #3263. Alternatively, why not simply providing two templates. Then at least these parts you already did in #3263 and what new people will add can use the common layout for validation errors. Furthermore, I also think that syntax/semantic validation distinction is pretty much still unanswered in doc/dev/error-categorization.md